dropwizard-metrics/dropwizard-metrics5/src/main/java/com/wavefront/integrations/metrics/WavefrontDropwizardReporter.java

```diff
@@ -0,0 +1,533 @@
+                            WavefrontSender wavefrontSender,
+                            final Clock clock,
+                            String prefix,
+                            String source,
+                            Map<String, String> pointTags,
+                            TimeUnit rateUnit,
+                            TimeUnit durationUnit,
+                            MetricFilter filter,
+                            boolean includeJvmMetrics,
+                            Set<MetricAttribute> disabledMetricAttributes) {
+    super(registry, "wavefront-reporter", filter, rateUnit, durationUnit, Executors.newSingleThreadScheduledExecutor(),
+        true, disabledMetricAttributes == null ? Collections.emptySet() : disabledMetricAttributes);
+    this.wavefront = wavefrontSender;
+    this.clock = clock;
+    this.prefix = prefix;
+    this.source = source;
+    this.pointTags = pointTags;
+
+    if (includeJvmMetrics) {
+      registry.register("jvm.uptime", (Gauge<Long>) () -> ManagementFactory.getRuntimeMXBean().getUptime());
+      registry.register("jvm.current_time", (Gauge<Long>) clock::getTime);
+      registry.register("jvm.classes", new ClassLoadingGaugeSet());
+      registry.register("jvm.fd_usage", new SafeFileDescriptorRatioGauge());
+      registry.register("jvm.buffers", new BufferPoolMetricSet(ManagementFactory.getPlatformMBeanServer()));
+      registry.register("jvm.gc", new GarbageCollectorMetricSet());
+      registry.register("jvm.memory", new MemoryUsageGaugeSet());
+      registry.register("jvm.thread-states", new ThreadStatesGaugeSet());
+    }
+  }
+
+  private WavefrontDropwizardReporter(MetricRegistry registry,
+                            String proxyHostname,
+                            int proxyPort,
+                            final Clock clock,
+                            String prefix,
+                            String source,
+                            Map<String, String> pointTags,
+                            TimeUnit rateUnit,
+                            TimeUnit durationUnit,
+                            MetricFilter filter,
+                            boolean includeJvmMetrics,
+                            Set<MetricAttribute> disabledMetricAttributes) {
+    this(registry, new Wavefront(proxyHostname, proxyPort), clock, prefix, source, pointTags, rateUnit,
+        durationUnit, filter, includeJvmMetrics, disabledMetricAttributes);
+  }
+
+  /**
+   * Called periodically by the polling thread. Subclasses should report all the given metrics.
+   *
+   * @param gauges     all of the gauges in the registry
+   * @param counters   all of the counters in the registry
+   * @param histograms all of the histograms in the registry
+   * @param meters     all of the meters in the registry
+   * @param timers     all of the timers in the registry
+   */
+  @Override
+  @SuppressWarnings("rawtypes")
+  public void report(SortedMap<MetricName, Gauge> gauges,
```

Comment:
Dropwizard 5 includes support for point tags on each metric level retrieved using `MetricName.getTags()`: https://github.com/dropwizard/metrics/blob/f75c56be7d1d71692a206dcd5a03a4427cffd8fb/metrics-core/src/main/java/io/dropwizard/metrics5/MetricName.java#L47

We should likely change all the report methods (reportCounter, reportGauge etc) to just take the MetricName as an argument and combine the metric level tags with the reporter level tags and pass along to the Sender.